packages/server/src/next-utils.ts

```diff
@@ -37,10 +37,10 @@ export async function nextStartDev(
   config: ServerConfig,
 ) {
   const transform = createOutputTransformer(manifest, devFolder).stream
-  const availablePort = await detect({port: config.port, hostname: config.hostname})
+  const availablePort = await detect({port: config.port!, hostname: config.hostname!})
```

Comment:
We don't know if port or hostname is set here. Probably a bit dangerous to use the non-null assertion operator (`!`). Be best if we check for undefined values and adjust accordingly or possibly tidy up the config to abstract away the differences between dev and build. Same as below

Reply:
@ryardley got it. it makes sense it add a conditional declaration rather than using an assertion operator. I will make the change. Thanks for the feedback